diff --git a/etc/config.sh/CGAL b/etc/config.sh/CGAL
index 68f679de00..8d7173d9fc 100644
--- a/etc/config.sh/CGAL
+++ b/etc/config.sh/CGAL
@@ -43,11 +43,11 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-boost_version=boost_1_74_0
-cgal_version=CGAL-4.14.3
+boost_version=boost-system
+cgal_version=CGAL-system
 
-export BOOST_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$boost_version"
-export CGAL_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$cgal_version"
+export BOOST_ARCH_PATH=$(brew --prefix boost)
+export CGAL_ARCH_PATH=$(brew --prefix cgal@4)
 
 # export GMP_ARCH_PATH=...
 # export MPFR_ARCH_PATH=...
diff --git a/etc/config.sh/FFTW b/etc/config.sh/FFTW
index 6d26f9d970..d63f0feacb 100644
--- a/etc/config.sh/FFTW
+++ b/etc/config.sh/FFTW
@@ -34,8 +34,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-fftw_version=fftw-3.3.10
-export FFTW_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$fftw_version
+fftw_version=fftw-system
+export FFTW_ARCH_PATH=$(brew --prefix fftw)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/adios2 b/etc/config.sh/adios2
index 4fe9d6183d..8e4897b89a 100644
--- a/etc/config.sh/adios2
+++ b/etc/config.sh/adios2
@@ -21,8 +21,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-adios2_version=ADIOS2-2.8.3
-export ADIOS2_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$adios2_version"
+adios2_version=ADIOS2-system
+export ADIOS2_ARCH_PATH=$(brew --prefix adios2)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/cmake b/etc/config.sh/cmake
index 617ee6dfc9..cac869fb9f 100644
--- a/etc/config.sh/cmake
+++ b/etc/config.sh/cmake
@@ -29,7 +29,7 @@
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
 cmake_version=cmake-system
-export CMAKE_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$cmake_version"
+export CMAKE_ARCH_PATH=$(brew --prefix cmake)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/hypre b/etc/config.sh/hypre
index 6f2648f2a3..d07bfdf0be 100644
--- a/etc/config.sh/hypre
+++ b/etc/config.sh/hypre
@@ -27,8 +27,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-hypre_version=hypre-2.23.0
-export HYPRE_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/$hypre_version"
+hypre_version=hypre-system
+export HYPRE_ARCH_PATH=$(brew --prefix hypre)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/kahip b/etc/config.sh/kahip
index ff7cec0813..098d446331 100644
--- a/etc/config.sh/kahip
+++ b/etc/config.sh/kahip
@@ -36,8 +36,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-KAHIP_VERSION=kahip-3.15
-export KAHIP_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$KAHIP_VERSION
+KAHIP_VERSION=kahip-system
+export KAHIP_ARCH_PATH=$(brew --prefix kahip)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/metis b/etc/config.sh/metis
index 6242dda174..32af9cb7b6 100644
--- a/etc/config.sh/metis
+++ b/etc/config.sh/metis
@@ -33,8 +33,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-METIS_VERSION=metis-5.1.0
-export METIS_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/$METIS_VERSION
+METIS_VERSION=metis-system
+export METIS_ARCH_PATH=$(brew --prefix metis)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/petsc b/etc/config.sh/petsc
index c512ebb376..41e441c3ab 100644
--- a/etc/config.sh/petsc
+++ b/etc/config.sh/petsc
@@ -27,8 +27,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-petsc_version=petsc-3.19.2
-export PETSC_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/$petsc_version"
+petsc_version=petsc-system
+export PETSC_ARCH_PATH=$(brew --prefix petsc)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/scotch b/etc/config.sh/scotch
index bb7261a059..4063f4c77a 100644
--- a/etc/config.sh/scotch
+++ b/etc/config.sh/scotch
@@ -35,8 +35,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-SCOTCH_VERSION=scotch_6.1.0
-export SCOTCH_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/$SCOTCH_VERSION
+SCOTCH_VERSION=scotch-system
+export SCOTCH_ARCH_PATH=$(brew --prefix scotch-no-pthread)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/settings b/etc/config.sh/settings
index 7735c4cf41..3eb37fdd02 100644
--- a/etc/config.sh/settings
+++ b/etc/config.sh/settings
@@ -72,6 +72,7 @@ Linux)
 # arm64 or x86_64 architectures
 Darwin)
     WM_ARCH=darwin64
+    export LIBOMP_PATH="$(brew --prefix libomp)"
     # Defult to clang (not gcc) as system compiler
     if [ "$WM_COMPILER" = Gcc ]
     then
diff --git a/src/OSspecific/POSIX/signals/feexceptErsatz.H b/src/OSspecific/POSIX/signals/feexceptErsatz.H
index e48b627758..d1f25226cf 100644
--- a/src/OSspecific/POSIX/signals/feexceptErsatz.H
+++ b/src/OSspecific/POSIX/signals/feexceptErsatz.H
@@ -38,20 +38,20 @@ inline int feenableexcept(unsigned int excepts)
         return -1;
     }
 
-#if defined __arm64__
-    old_excepts = fenv.__fpcr & FE_ALL_EXCEPT;
-#else
+    #if defined __arm64__
+    old_excepts = fenv.__fpsr & FE_ALL_EXCEPT;
+    #else
     old_excepts = fenv.__control & FE_ALL_EXCEPT;
-#endif
+    #endif
 
     // unmask
-#if defined __arm64__
-    fenv.__fpcr &= ~new_excepts;
-    fenv.__fpsr &= ~(new_excepts << 7);
-#else
+    #if defined __arm64__
+    fenv.__fpsr |= new_excepts;
+    fenv.__fpcr |= new_excepts << 8;
+    #else
     fenv.__control &= ~new_excepts;
     fenv.__mxcsr   &= ~(new_excepts << 7);
-#endif
+    #endif
 
     return fesetenv(&fenv) ? -1 : old_excepts;
 }
@@ -68,20 +68,20 @@ inline int fedisableexcept(unsigned int excepts)
         return -1;
     }
 
-#if defined __arm64__
-    old_excepts = fenv.__fpcr & FE_ALL_EXCEPT;
-#else
+    #if defined __arm64__
+    old_excepts = fenv.__fpsr & FE_ALL_EXCEPT;
+    #else
     old_excepts = fenv.__control & FE_ALL_EXCEPT;
-#endif
+    #endif
 
     // mask
-#if defined __arm64__
-    fenv.__fpcr |= new_excepts;
-    fenv.__fpsr |= new_excepts << 7;
-#else
+    #if defined __arm64__
+    fenv.__fpsr &= ~new_excepts;
+    fenv.__fpcr &= ~(new_excepts << 8);
+    #else
     fenv.__control |= new_excepts;
     fenv.__mxcsr   |= new_excepts << 7;
-#endif
+    #endif
 
     return fesetenv(&fenv) ? -1 : old_excepts;
 }
diff --git a/src/OSspecific/POSIX/signals/sigFpe.C b/src/OSspecific/POSIX/signals/sigFpe.C
index c02588e19b..2824f63bcc 100644
--- a/src/OSspecific/POSIX/signals/sigFpe.C
+++ b/src/OSspecific/POSIX/signals/sigFpe.C
@@ -118,9 +118,12 @@ void* Foam::sigFpe::mallocNan(size_t size)
 void Foam::sigFpe::sigHandler(int)
 {
     #if (defined(__linux__) && defined(__GNUC__)) || defined(__APPLE__)
-
+    // Workaround for Apple Silicon
+    #if defined(__APPLE__) && defined(__arm64__)
+    resetHandler("SIGFPE", SIGILL);
+    #else
     resetHandler("SIGFPE", SIGFPE);
-
+    #endif
     JobInfo::shutdown();        // From running -> finished
     error::printStack(Perr);
     ::raise(SIGFPE);            // Throw signal (to old handler)
@@ -192,8 +195,12 @@ void Foam::sigFpe::set(bool verbose)
           | FE_INVALID
           | FE_OVERFLOW
         );
-
+        // Workaround for Apple Silicon
+        #if defined(__APPLE__) && defined(__arm64__)
+        setHandler("SIGFPE", SIGILL, sigHandler);
+        #else
         setHandler("SIGFPE", SIGFPE, sigHandler);
+        #endif
 
         sigActive_ = true;
         #endif
@@ -248,8 +255,12 @@ void Foam::sigFpe::unset(bool verbose)
             Info<< "sigFpe : Disabling floating point exception trapping"
                 << endl;
         }
-
+        // Workaround for Apple Silicon
+        #if defined(__APPLE__) && defined(__arm64__)
+        resetHandler("SIGFPE", SIGILL);
+        #else
         resetHandler("SIGFPE", SIGFPE);
+        #endif
 
         // Reset exception raising
         const int oldExcept = fedisableexcept
diff --git a/wmake/rules/darwin64Clang/c b/wmake/rules/darwin64Clang/c
index d145bd5b92..9e40bdef57 100644
--- a/wmake/rules/darwin64Clang/c
+++ b/wmake/rules/darwin64Clang/c
@@ -1,7 +1,7 @@
 #------------------------------------------------------------------------------
 include $(GENERAL_RULES)/Clang/c
 
-cARCH      := -m64 -ftrapping-math
+cARCH      := -ftrapping-math -ffp-contract=off
 
 ifneq (,$(strip $(WM_COMPILE_OPTION)))
     sinclude $(DEFAULT_RULES)/c$(WM_COMPILE_OPTION)
@@ -15,8 +15,8 @@ ctoo        = $(WM_SCHEDULER) $(cc) $(cFLAGS) -c $< -o $@
 
 LINK_LIBS   = $(cDBUG)
 
-LINKLIBSO   = $(cc) $(cARCH) -Wl,-dylib,-undefined,dynamic_lookup
+LINKLIBSO   = $(cc) $(cARCH) -Wl,-undefined,dynamic_lookup,-no_fixup_chains,-dylib
 
-LINKEXE     = $(cc) $(cARCH) -Wl,-execute,-undefined,dynamic_lookup
+LINKEXE     = $(cc) $(cARCH) -Wl,-undefined,dynamic_lookup,-no_fixup_chains,-execute
 
 #------------------------------------------------------------------------------
diff --git a/wmake/rules/darwin64Clang/c++ b/wmake/rules/darwin64Clang/c++
index 7547b28d53..0ca8df4660 100644
--- a/wmake/rules/darwin64Clang/c++
+++ b/wmake/rules/darwin64Clang/c++
@@ -1,7 +1,7 @@
 #------------------------------------------------------------------------------
 include $(GENERAL_RULES)/Clang/c++
 
-c++ARCH    := -m64 -pthread -ftrapping-math
+c++ARCH    := -pthread -ftrapping-math -ffp-contract=off
 
 ifneq (,$(strip $(WM_COMPILE_OPTION)))
     sinclude $(DEFAULT_RULES)/c++$(WM_COMPILE_OPTION)
@@ -18,8 +18,8 @@ cxxtoo      = $(Ctoo)
 
 LINK_LIBS   = $(c++DBUG)
 
-LINKLIBSO   = $(CC) $(c++FLAGS) -Wl,-dylib,-undefined,dynamic_lookup
+LINKLIBSO   = $(CC) $(c++FLAGS) -Wl,-undefined,dynamic_lookup,-no_fixup_chains,-dylib
 
-LINKEXE     = $(CC) $(c++FLAGS) -Wl,-execute,-undefined,dynamic_lookup
+LINKEXE     = $(CC) $(c++FLAGS) -Wl,-undefined,dynamic_lookup,-no_fixup_chains,-execute
 
 #------------------------------------------------------------------------------
diff --git a/wmake/rules/darwin64Clang/general b/wmake/rules/darwin64Clang/general
index 4a5acb5ef9..c5c0156d86 100644
--- a/wmake/rules/darwin64Clang/general
+++ b/wmake/rules/darwin64Clang/general
@@ -5,8 +5,8 @@ include $(GENERAL_RULES)/standard
 
 # Darwin-specific
 # ----
-COMP_OPENMP = -Xpreprocessor -fopenmp
-LINK_OPENMP = -lomp
+COMP_OPENMP = -Xpreprocessor -fopenmp -I$(LIBOMP_PATH)/include
+LINK_OPENMP = -L$(LIBOMP_PATH)/lib -lomp
 # ----
 
 ifneq (,$(findstring ~openmp,$(WM_COMPILE_CONTROL)))
