diff --git a/etc/bashrc b/etc/bashrc
index 9d7b3effaa..99e4056d6f 100644
--- a/etc/bashrc
+++ b/etc/bashrc
@@ -68,7 +68,7 @@ export WM_COMPILER_TYPE=system
 # [WM_COMPILER] - Compiler:
 # = Gcc | Clang | Icc | Icx | Amd | Arm | Cray | Fujitsu | Nvidia |
 #   Gcc<digits> | Clang<digits>
-export WM_COMPILER=Gcc
+export WM_COMPILER=Clang
 
 # [WM_PRECISION_OPTION] - Floating-point precision:
 # = DP | SP | SPDP
diff --git a/etc/config.sh/CGAL b/etc/config.sh/CGAL
index 68f679de00..fdea6655f2 100644
--- a/etc/config.sh/CGAL
+++ b/etc/config.sh/CGAL
@@ -43,11 +43,11 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-boost_version=boost_1_74_0
-cgal_version=CGAL-4.14.3
+boost_version=boost-system
+cgal_version=CGAL-system
 
-export BOOST_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$boost_version"
-export CGAL_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$cgal_version"
+export BOOST_ARCH_PATH=$(brew --prefix)/opt/boost
+export CGAL_ARCH_PATH=$(brew --prefix)/opt/cgal@4
 
 # export GMP_ARCH_PATH=...
 # export MPFR_ARCH_PATH=...
diff --git a/etc/config.sh/FFTW b/etc/config.sh/FFTW
index 6d26f9d970..4305f9b3f3 100644
--- a/etc/config.sh/FFTW
+++ b/etc/config.sh/FFTW
@@ -34,8 +34,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-fftw_version=fftw-3.3.10
-export FFTW_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$fftw_version
+fftw_version=fftw-system
+export FFTW_ARCH_PATH=$(brew --prefix)/opt/fftw
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/adios2 b/etc/config.sh/adios2
index 7be5374b0c..ef57587de9 100644
--- a/etc/config.sh/adios2
+++ b/etc/config.sh/adios2
@@ -21,8 +21,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-adios2_version=ADIOS2-2.7.1
-export ADIOS2_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$adios2_version"
+adios2_version=ADIOS2-system
+export ADIOS2_ARCH_PATH=$(brew --prefix)/opt/adios2
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/cmake b/etc/config.sh/cmake
index 617ee6dfc9..9e1beee716 100644
--- a/etc/config.sh/cmake
+++ b/etc/config.sh/cmake
@@ -29,7 +29,7 @@
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
 cmake_version=cmake-system
-export CMAKE_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$cmake_version"
+export CMAKE_ARCH_PATH=$(brew --prefix)/opt/cmake
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/kahip b/etc/config.sh/kahip
index 135c4e2dd0..83f7b54b71 100644
--- a/etc/config.sh/kahip
+++ b/etc/config.sh/kahip
@@ -36,8 +36,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-KAHIP_VERSION=kahip-3.14
-export KAHIP_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$KAHIP_VERSION
+KAHIP_VERSION=kahip-system
+export KAHIP_ARCH_PATH=$(brew --prefix)/opt/kahip
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/metis b/etc/config.sh/metis
index 6242dda174..34597e8044 100644
--- a/etc/config.sh/metis
+++ b/etc/config.sh/metis
@@ -33,8 +33,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-METIS_VERSION=metis-5.1.0
-export METIS_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/$METIS_VERSION
+METIS_VERSION=metis-system
+export METIS_ARCH_PATH=$(brew --prefix)/opt/metis
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/petsc b/etc/config.sh/petsc
index 265222d262..e53341dfae 100644
--- a/etc/config.sh/petsc
+++ b/etc/config.sh/petsc
@@ -27,8 +27,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-petsc_version=petsc-3.17.2
-export PETSC_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/$petsc_version"
+petsc_version=petsc-system
+export PETSC_ARCH_PATH=$(brew --prefix)/opt/petsc
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/scotch b/etc/config.sh/scotch
index 78a56d9d92..4561d9c8dc 100644
--- a/etc/config.sh/scotch
+++ b/etc/config.sh/scotch
@@ -33,8 +33,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-SCOTCH_VERSION=scotch_6.1.0
-export SCOTCH_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/$SCOTCH_VERSION
+SCOTCH_VERSION=scotch-system
+export SCOTCH_ARCH_PATH=$(brew --prefix)/opt/scotch-no-pthread
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/settings b/etc/config.sh/settings
index 8024eb7519..d79f745ec0 100644
--- a/etc/config.sh/settings
+++ b/etc/config.sh/settings
@@ -80,13 +80,19 @@ Linux)
 
 # arm64 or x86_64 architectures
 Darwin)
-    WM_ARCH=darwin64
-    # Defult to clang (not gcc) as system compiler
-    if [ "$WM_COMPILER" = Gcc ]
-    then
-        WM_COMPILER=Clang
-        echo "openfoam (darwin): using clang instead of gcc" 1>&2
-    fi
+    case "$(uname -m)" in
+    x86_64)
+        # Presume x86_64
+        WM_ARCH=darwin64
+        ;;
+    arm64)
+        # Presume Apple silicon
+        WM_ARCH=darwinARM64
+        ;;
+    *)
+        echo "openfoam: ($(uname -m)) - unknown processor type" 1>&2
+        ;;
+    esac
     ;;
 
 # Presume x86_64, with mingw cross-compiled
diff --git a/src/meshTools/triSurface/triSurfaceTools/geompack/geompack.C b/src/meshTools/triSurface/triSurfaceTools/geompack/geompack.C
index 7298db6060..b586baf85d 100644
--- a/src/meshTools/triSurface/triSurfaceTools/geompack/geompack.C
+++ b/src/meshTools/triSurface/triSurfaceTools/geompack/geompack.C
@@ -6,6 +6,10 @@
 # include <ctime>
 # include <cstring>
 
+#if defined(__APPLE__) && defined(__clang__)
+#pragma clang fp exceptions(ignore)
+#endif
+
 using namespace std;
 
 # include "geompack.H"
diff --git a/wmake/rules/darwin64Clang/c b/wmake/rules/darwin64Clang/c
index 70c7e06576..dad9ad723b 100644
--- a/wmake/rules/darwin64Clang/c
+++ b/wmake/rules/darwin64Clang/c
@@ -14,8 +14,8 @@ ctoo        = $(WM_SCHEDULER) $(cc) $(cFLAGS) -c $< -o $@
 
 LINK_LIBS   = $(cDBUG)
 
-LINKLIBSO   = $(cc) $(cARCH) -Wl,-dylib,-undefined,dynamic_lookup
+LINKLIBSO   = $(cc) $(cARCH) -Wl,-w,-dylib,-undefined,dynamic_lookup
 
-LINKEXE     = $(cc) $(cARCH) -Wl,-execute,-undefined,dynamic_lookup
+LINKEXE     = $(cc) $(cARCH) -Wl,-w,-execute,-undefined,dynamic_lookup
 
 #------------------------------------------------------------------------------
diff --git a/wmake/rules/darwin64Clang/c++ b/wmake/rules/darwin64Clang/c++
index 8c2eeba107..ae95f51c8f 100644
--- a/wmake/rules/darwin64Clang/c++
+++ b/wmake/rules/darwin64Clang/c++
@@ -16,8 +16,8 @@ cxxtoo      = $(Ctoo)
 
 LINK_LIBS   = $(c++DBUG)
 
-LINKLIBSO   = $(CC) $(c++FLAGS) -Wl,-dylib,-undefined,dynamic_lookup
+LINKLIBSO   = $(CC) $(c++FLAGS) -Wl,-w,-dylib,-undefined,dynamic_lookup
 
-LINKEXE     = $(CC) $(c++FLAGS) -Wl,-execute,-undefined,dynamic_lookup
+LINKEXE     = $(CC) $(c++FLAGS) -Wl,-w,-execute,-undefined,dynamic_lookup
 
 #------------------------------------------------------------------------------
diff --git a/wmake/rules/darwin64Clang/general b/wmake/rules/darwin64Clang/general
index c305312685..88dc9ab61b 100644
--- a/wmake/rules/darwin64Clang/general
+++ b/wmake/rules/darwin64Clang/general
@@ -5,8 +5,8 @@ include $(GENERAL_RULES)/standard
 
 # Darwin-specific
 # ----
-COMP_OPENMP = -DUSE_OMP -Xpreprocessor -fopenmp
-LINK_OPENMP = -lomp
+COMP_OPENMP = -DUSE_OMP -Xpreprocessor -fopenmp -I/usr/local/opt/libomp/include
+LINK_OPENMP = -L/usr/local/opt/libomp/lib -lomp
 # ----
 
 ifneq (,$(findstring ~openmp,$(WM_COMPILE_CONTROL)))
diff --git a/wmake/rules/darwinARM64Clang/c b/wmake/rules/darwinARM64Clang/c
new file mode 100644
index 0000000000..cd9c6ff9e7
--- /dev/null
+++ b/wmake/rules/darwinARM64Clang/c
@@ -0,0 +1,23 @@
+#------------------------------------------------------------------------------
+include $(GENERAL_RULES)/Clang/c
+
+c           = clang
+
+cARCH       =
+
+# Compile option is non-mandatory, but must be non-empty
+sinclude $(DEFAULT_RULES)/c$(WM_COMPILE_OPTION)
+
+cFLAGS      = \
+    $(cARCH) $(GFLAGS) $(cWARN) $(cOPT) $(cDBUG) \
+    $(FOAM_EXTRA_CFLAGS) $(LIB_HEADER_DIRS) -fPIC
+
+ctoo        = $(WM_SCHEDULER) $(cc) $(cFLAGS) -c $< -o $@
+
+LINK_LIBS   = $(cDBUG)
+
+LINKLIBSO   = $(cc) $(cARCH) -Wl,-w,-dylib,-undefined,dynamic_lookup
+
+LINKEXE     = $(cc) $(cARCH) -Wl,-w,-execute,-undefined,dynamic_lookup
+
+#------------------------------------------------------------------------------
diff --git a/wmake/rules/darwinARM64Clang/c++ b/wmake/rules/darwinARM64Clang/c++
new file mode 100644
index 0000000000..3bf10d58b0
--- /dev/null
+++ b/wmake/rules/darwinARM64Clang/c++
@@ -0,0 +1,25 @@
+#------------------------------------------------------------------------------
+include $(GENERAL_RULES)/Clang/c++
+
+CC          = clang++
+
+c++ARCH     = -std=c++14 -pthread
+
+include $(DEFAULT_RULES)/c++$(WM_COMPILE_OPTION)
+
+c++FLAGS    = \
+    $(c++ARCH) $(GFLAGS) $(c++WARN) $(c++OPT) $(c++DBUG) $(ptFLAGS) \
+    $(FOAM_EXTRA_CXXFLAGS) $(LIB_HEADER_DIRS) -fPIC
+
+Ctoo        = $(WM_SCHEDULER) $(CC) $(c++FLAGS) -c $< -o $@
+cctoo       = $(Ctoo)
+cpptoo      = $(Ctoo)
+cxxtoo      = $(Ctoo)
+
+LINK_LIBS   = $(c++DBUG)
+
+LINKLIBSO   = $(CC) $(c++FLAGS) -Wl,-w,-dylib,-undefined,dynamic_lookup
+
+LINKEXE     = $(CC) $(c++FLAGS) -Wl,-w,-execute,-undefined,dynamic_lookup
+
+#------------------------------------------------------------------------------
diff --git a/wmake/rules/darwinARM64Clang/c++Debug b/wmake/rules/darwinARM64Clang/c++Debug
new file mode 100644
index 0000000000..48f0b6643e
--- /dev/null
+++ b/wmake/rules/darwinARM64Clang/c++Debug
@@ -0,0 +1,2 @@
+c++DBUG     = -g -DFULLDEBUG
+c++OPT      = -O0
diff --git a/wmake/rules/darwinARM64Clang/c++Opt b/wmake/rules/darwinARM64Clang/c++Opt
new file mode 100644
index 0000000000..2aedabd628
--- /dev/null
+++ b/wmake/rules/darwinARM64Clang/c++Opt
@@ -0,0 +1,2 @@
+c++DBUG     =
+c++OPT      = -O3
diff --git a/wmake/rules/darwinARM64Clang/c++Prof b/wmake/rules/darwinARM64Clang/c++Prof
new file mode 100644
index 0000000000..3bda4dad55
--- /dev/null
+++ b/wmake/rules/darwinARM64Clang/c++Prof
@@ -0,0 +1,2 @@
+c++DBUG    = -pg
+c++OPT     = -O2
diff --git a/wmake/rules/darwinARM64Clang/cDebug b/wmake/rules/darwinARM64Clang/cDebug
new file mode 100644
index 0000000000..7b7adf10de
--- /dev/null
+++ b/wmake/rules/darwinARM64Clang/cDebug
@@ -0,0 +1,2 @@
+cDBUG       = -g -DFULLDEBUG
+cOPT        = -O0
diff --git a/wmake/rules/darwinARM64Clang/cOpt b/wmake/rules/darwinARM64Clang/cOpt
new file mode 100644
index 0000000000..17318709f1
--- /dev/null
+++ b/wmake/rules/darwinARM64Clang/cOpt
@@ -0,0 +1,2 @@
+cDBUG       =
+cOPT        = -O3
diff --git a/wmake/rules/darwinARM64Clang/cProf b/wmake/rules/darwinARM64Clang/cProf
new file mode 100644
index 0000000000..ca3ac9bf5f
--- /dev/null
+++ b/wmake/rules/darwinARM64Clang/cProf
@@ -0,0 +1,2 @@
+cDBUG       = -pg
+cOPT        = -O2
diff --git a/wmake/rules/darwinARM64Clang/cgal b/wmake/rules/darwinARM64Clang/cgal
new file mode 100644
index 0000000000..2f376af1e1
--- /dev/null
+++ b/wmake/rules/darwinARM64Clang/cgal
@@ -0,0 +1,14 @@
+# ----------------------------------------------------------------------------
+# CGAL on Darwin
+# CGAL (library version) without mpfr
+
+CGAL_INC  = \
+    $(foreach dir,$(BOOST_INC_DIR),-I$(dir)) \
+    $(foreach dir,$(CGAL_INC_DIR),-I$(dir))
+
+CGAL_LIBS = \
+    $(foreach dir,$(BOOST_LIB_DIR),-L$(dir)) \
+    $(foreach dir,$(CGAL_LIB_DIR),-L$(dir)) \
+    -lCGAL
+
+# ----------------------------------------------------------------------------
diff --git a/wmake/rules/darwinARM64Clang/general b/wmake/rules/darwinARM64Clang/general
new file mode 100644
index 0000000000..3a88c37947
--- /dev/null
+++ b/wmake/rules/darwinARM64Clang/general
@@ -0,0 +1,22 @@
+CPP        = cpp -traditional-cpp $(GFLAGS)
+
+include $(GENERAL_RULES)/standard
+## include $(GENERAL_RULES)/Clang/openmp
+
+# Darwin-specific
+# ----
+COMP_OPENMP = -DUSE_OMP -Xpreprocessor -fopenmp -I/opt/homebrew/opt/libomp/include
+LINK_OPENMP = -L/opt/homebrew/opt/libomp/lib -lomp
+# ----
+
+ifneq (,$(findstring ~openmp,$(WM_COMPILE_CONTROL)))
+    include $(GENERAL_RULES)/no-openmp
+endif
+
+include $(DEFAULT_RULES)/c
+include $(DEFAULT_RULES)/c++
+
+# Shared library extension (with '.' separator)
+EXT_SO  = .dylib
+
+# -----------------------------------------------------------------------------
