diff --git a/etc/config.sh/CGAL b/etc/config.sh/CGAL
index 68f679de00..1bf1e889de 100644
--- a/etc/config.sh/CGAL
+++ b/etc/config.sh/CGAL
@@ -43,14 +43,14 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-boost_version=boost_1_74_0
-cgal_version=CGAL-4.14.3
+boost_version=boost-system
+cgal_version=CGAL-system
 
-export BOOST_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$boost_version"
-export CGAL_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$cgal_version"
+export BOOST_ARCH_PATH=$(brew --prefix boost)
+export CGAL_ARCH_PATH=$(brew --prefix cgal)
 
-# export GMP_ARCH_PATH=...
-# export MPFR_ARCH_PATH=...
+export GMP_ARCH_PATH=$(brew --prefix gmp)
+export MPFR_ARCH_PATH=$(brew --prefix mpfr)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/FFTW b/etc/config.sh/FFTW
index 6d26f9d970..d63f0feacb 100644
--- a/etc/config.sh/FFTW
+++ b/etc/config.sh/FFTW
@@ -34,8 +34,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-fftw_version=fftw-3.3.10
-export FFTW_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$fftw_version
+fftw_version=fftw-system
+export FFTW_ARCH_PATH=$(brew --prefix fftw)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/adios2 b/etc/config.sh/adios2
index ae967dc7c8..1eac6d4caa 100644
--- a/etc/config.sh/adios2
+++ b/etc/config.sh/adios2
@@ -21,8 +21,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-adios2_version=ADIOS2-2.9.2
-export ADIOS2_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$adios2_version"
+adios2_version=ADIOS2-system
+export ADIOS2_ARCH_PATH=$(brew --prefix adios2)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/cmake b/etc/config.sh/cmake
index 617ee6dfc9..cac869fb9f 100644
--- a/etc/config.sh/cmake
+++ b/etc/config.sh/cmake
@@ -29,7 +29,7 @@
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
 cmake_version=cmake-system
-export CMAKE_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$cmake_version"
+export CMAKE_ARCH_PATH=$(brew --prefix cmake)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/hypre b/etc/config.sh/hypre
index 6f2648f2a3..d07bfdf0be 100644
--- a/etc/config.sh/hypre
+++ b/etc/config.sh/hypre
@@ -27,8 +27,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-hypre_version=hypre-2.23.0
-export HYPRE_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/$hypre_version"
+hypre_version=hypre-system
+export HYPRE_ARCH_PATH=$(brew --prefix hypre)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/kahip b/etc/config.sh/kahip
index ff7cec0813..098d446331 100644
--- a/etc/config.sh/kahip
+++ b/etc/config.sh/kahip
@@ -36,8 +36,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-KAHIP_VERSION=kahip-3.15
-export KAHIP_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$KAHIP_VERSION
+KAHIP_VERSION=kahip-system
+export KAHIP_ARCH_PATH=$(brew --prefix kahip)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/metis b/etc/config.sh/metis
index 6242dda174..32af9cb7b6 100644
--- a/etc/config.sh/metis
+++ b/etc/config.sh/metis
@@ -33,8 +33,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-METIS_VERSION=metis-5.1.0
-export METIS_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/$METIS_VERSION
+METIS_VERSION=metis-system
+export METIS_ARCH_PATH=$(brew --prefix metis)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/petsc b/etc/config.sh/petsc
index c512ebb376..41e441c3ab 100644
--- a/etc/config.sh/petsc
+++ b/etc/config.sh/petsc
@@ -27,8 +27,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-petsc_version=petsc-3.19.2
-export PETSC_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/$petsc_version"
+petsc_version=petsc-system
+export PETSC_ARCH_PATH=$(brew --prefix petsc)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/scotch b/etc/config.sh/scotch
index bb7261a059..6dd6202f41 100644
--- a/etc/config.sh/scotch
+++ b/etc/config.sh/scotch
@@ -35,8 +35,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-SCOTCH_VERSION=scotch_6.1.0
-export SCOTCH_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/$SCOTCH_VERSION
+SCOTCH_VERSION=scotch-system
+export SCOTCH_ARCH_PATH=$(brew --prefix scotch)
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
diff --git a/etc/config.sh/settings b/etc/config.sh/settings
index 7735c4cf41..3eb37fdd02 100644
--- a/etc/config.sh/settings
+++ b/etc/config.sh/settings
@@ -72,6 +72,7 @@ Linux)
 # arm64 or x86_64 architectures
 Darwin)
     WM_ARCH=darwin64
+    export LIBOMP_PATH="$(brew --prefix libomp)"
     # Defult to clang (not gcc) as system compiler
     if [ "$WM_COMPILER" = Gcc ]
     then
diff --git a/etc/config.sh/setup b/etc/config.sh/setup
index 837eb90de4..bf9c7c22e3 100644
--- a/etc/config.sh/setup
+++ b/etc/config.sh/setup
@@ -230,7 +230,8 @@ then
 fi
 
 # OpenFOAM user, group libraries
-_foamAddLib  "$FOAM_USER_LIBBIN:$FOAM_SITE_LIBBIN"
+_foamAddLib  "$FOAM_SITE_LIBBIN"
+_foamAddLib  "$FOAM_USER_LIBBIN"
 
 if [ -d "$WM_PROJECT_DIR/doc/man1" ]
 then
diff --git a/src/OpenFOAM/db/IOstreams/memory/memoryStreamBuffer.H b/src/OpenFOAM/db/IOstreams/memory/memoryStreamBuffer.H
index 8de6992f87..ea9ef6abe0 100644
--- a/src/OpenFOAM/db/IOstreams/memory/memoryStreamBuffer.H
+++ b/src/OpenFOAM/db/IOstreams/memory/memoryStreamBuffer.H
@@ -519,7 +519,8 @@ public:
                 if (newCapacity < len)
                 {
                     // Increase capacity (doubling)
-                    newCapacity = max(len, label(2*storage_.size()));
+                    newCapacity =
+                        Foam::max(label(len), label(2*storage_.size()));
                 }
 
                 // Info<<"request:" << len
diff --git a/wmake/rules/darwin64Clang/c b/wmake/rules/darwin64Clang/c
index 3083b281ed..4648a0dcd3 100644
--- a/wmake/rules/darwin64Clang/c
+++ b/wmake/rules/darwin64Clang/c
@@ -5,7 +5,7 @@ ifneq (,$(findstring +xcrun,$(WM_COMPILE_CONTROL)))
 cc         := xcrun cc
 endif
 
-cARCH      := -m64 -ftrapping-math
+cARCH      := -m64 -ftrapping-math -ffp-contract=off
 
 ifneq (,$(strip $(WM_COMPILE_OPTION)))
     sinclude $(DEFAULT_RULES)/c$(WM_COMPILE_OPTION)
diff --git a/wmake/rules/darwin64Clang/c++ b/wmake/rules/darwin64Clang/c++
index 01158bcb0f..099d1fda12 100644
--- a/wmake/rules/darwin64Clang/c++
+++ b/wmake/rules/darwin64Clang/c++
@@ -5,7 +5,7 @@ ifneq (,$(findstring +xcrun,$(WM_COMPILE_CONTROL)))
 CC         := xcrun c++ -std=c++14
 endif
 
-c++ARCH    := -m64 -pthread -ftrapping-math
+c++ARCH    := -m64 -pthread -ftrapping-math -ffp-contract=off
 
 ifneq (,$(strip $(WM_COMPILE_OPTION)))
     sinclude $(DEFAULT_RULES)/c++$(WM_COMPILE_OPTION)
diff --git a/wmake/rules/darwin64Clang/general b/wmake/rules/darwin64Clang/general
index 4a5acb5ef9..c5c0156d86 100644
--- a/wmake/rules/darwin64Clang/general
+++ b/wmake/rules/darwin64Clang/general
@@ -5,8 +5,8 @@ include $(GENERAL_RULES)/standard
 
 # Darwin-specific
 # ----
-COMP_OPENMP = -Xpreprocessor -fopenmp
-LINK_OPENMP = -lomp
+COMP_OPENMP = -Xpreprocessor -fopenmp -I$(LIBOMP_PATH)/include
+LINK_OPENMP = -L$(LIBOMP_PATH)/lib -lomp
 # ----
 
 ifneq (,$(findstring ~openmp,$(WM_COMPILE_CONTROL)))
